src/Transports.Subscriptions.WebSockets/WebsocketWriterStream.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.IO;
+using System.Net.WebSockets;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace GraphQL.Server.Transports.WebSockets
+{
+    public class WebsocketWriterStream : Stream
+    {
+        private readonly WebSocket _webSocket;
+
+        public WebsocketWriterStream(WebSocket webSocket)
+        {
+            _webSocket = webSocket;
+        }
+
+        public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        {
+            return _webSocket.SendAsync(new ArraySegment<byte>(buffer, offset, count), WebSocketMessageType.Text, false,
+                cancellationToken);
+        }
+
+        public override Task FlushAsync(CancellationToken cancellationToken)
+        {
+            return _webSocket.SendAsync(new ArraySegment<byte>(Array.Empty<byte>()), WebSocketMessageType.Text, true, cancellationToken);
+        }
+
+        public override void Flush()
+        {
+            throw new System.NotImplementedException("Synchronous methods are not supported by WebsocketWriterStream.");
+        }
+
+        public override int Read(byte[] buffer, int offset, int count)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public override long Seek(long offset, SeekOrigin origin)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public override void SetLength(long value)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public override void Write(byte[] buffer, int offset, int count)
+        {
+            throw new System.NotImplementedException("Synchronous methods are not supported by WebsocketWriterStream.");
```

Comment:
I believe `JsonTextWriter` will end up calling the synchronous `Write()` method, so I think this will need to be supported.

Reply:
The implementation from Asp.Net Core Mvc (HttpResponseStreamWriter) should always use async method (at least in theory, in my tests, sync methods were never called). 
But I do agree that sync write and flush methods should be implemented (especially considering that alternative DocumentWriter implementation can be used). Perhaps it's worth to add a warning message to logs to notify the users that sync method was used as well.

Reply:
After taking a second look at the code, the fully async flow will only be used if the response is smaller than StreamWriter buffer. Implementing sync method is definitely required.

Reply:
Yeah, that's why I think unit tests would be good. With different size responses.